### PR TITLE
Additional tests for custom sdk scenario

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2069 followup

## What
- make documentation internal, remove `CHANGELOG.md` entry
- apply suggestions from https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2069#discussion_r1087430686
- add metrics tests

## Tests

Included in PR.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

**Review by pellared — APPROVED** [review]

File: test/test-applications/integrations/TestApplication.CustomSdk/Program.cs [hunk 1]
```diff
@@ -29,14 +32,39 @@ public static class Program
...
+
+        using (var activity = ActivitySource.StartActivity("Manual"))
+        {
+            await PingRedis(args);
+
+            using var client = new HttpClient();
+            await client.GetStringAsync("https://www.bing.com");
+        }
+
+        Thread.Sleep(TimeSpan.FromSeconds(10));
```
Comment (pjanotti) [inline test/test-applications/integrations/TestApplication.CustomSdk/Program.cs:62]: Do we really need this sleep here? It is also quite long.

**Review by pjanotti — APPROVED** [review]
@lachmatt could you please see if the sleep could be removed or at least reduced?

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,7 +18,6 @@ This beta release is built on top of [OpenTelemetry .NET](https://github.com/ope
 - Support configuring `OTEL_*` settings using `App.config` and `Web.config`.
 - Add support for Quartz traces instrumentation.
 - Add support for EntityFrameworkCore traces instrumentations.
-- Add support for custom OpenTelemetry .NET SDK through `OTEL_DOTNET_AUTO_SETUP_SDK`.
```
Comment (rajkumar-rangaraj) [inline CHANGELOG.md:21]: Why this is removed from changelog?

**Review by Kielek — APPROVED** [review]

File: test/test-applications/integrations/TestApplication.CustomSdk/Program.cs (hunk 1 above)
Comment (pellared) [reply test/test-applications/integrations/TestApplication.CustomSdk/Program.cs:62]: I think it is not needed...

Comment (lachmatt) [reply test/test-applications/integrations/TestApplication.CustomSdk/Program.cs]: Changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2081/commits/817994092dd4edee8dbefedf3a41ca148abab5d2 in order for it not to be needed anymore

**Review by pellared — APPROVED** [review]

**Pull request merged**